test/extensions/filters/http/health_check/health_check_test.cc

```diff
@@ -65,16 +78,8 @@ class HealthCheckFilterTest : public testing::Test {
   Http::TestRequestHeaderMapImpl request_headers_;
   Http::TestRequestHeaderMapImpl request_headers_no_hc_;
   HeaderDataVectorSharedPtr header_data_;
-
-  class MockHealthCheckCluster : public NiceMock<Upstream::MockThreadLocalCluster> {
-  public:
-    MockHealthCheckCluster(uint64_t membership_total, uint64_t membership_healthy,
-                           uint64_t membership_degraded = 0) {
-      info()->endpointStats().membership_total_.set(membership_total);
-      info()->endpointStats().membership_healthy_.set(membership_healthy);
-      info()->endpointStats().membership_degraded_.set(membership_degraded);
-    }
-  };
+  Stats::TestUtil::TestStore stats_store_;
+  HealthCheckFilterStatsSharedPtr stats_;
```

Comment:
nit: I may be missing something, but I don't see this is being used. Consider removing it from the data member and add it directly to `prepareFilter()`.